src/main/java/com/oath/halodb/HaloDBInternal.java

```diff
@@ -397,20 +406,22 @@ private int getNextFileId() {
             .max(Comparator.comparingInt(HaloDBFile::getFileId));
     }
 
-    private void buildInMemoryIndex(HaloDBOptions options) throws IOException {
+    private long buildInMemoryIndex(HaloDBOptions options) throws IOException {
         //TODO: probably processing files in descending order is more efficient.
         List<Integer> indexFiles = FileUtils.listIndexFiles(dbDirectory);
 
         logger.info("About to scan {} index files to construct index ...", indexFiles.size());
 
         long start = System.currentTimeMillis();
 
+        long maxSequenceNumber = -1l;
```

Comment:
minor change: could you please move this declaration right after `long start` without the blank line.

Reply:
done